stdlib/socketserver.pyi

```diff
@@ -6,17 +6,19 @@ from typing import Any, BinaryIO, Callable, ClassVar, Union
 
 _RequestType = Union[_socket, tuple[bytes, _socket]]
 _AddressType = Union[tuple[str, int], str]
+_RequestHandlerClassCallable = Callable[[Any, Any, BaseServer], BaseRequestHandler]
```

Comment:
I don't like the hard-coded `BaseServer` here. If you are making `WebServer` that inherits from `TCPServer`, your handler should accept a `WebServer` as the third argument. It shouldn't have to work with any `BaseServer` or even any `TcpServer`.

You can use `Self` to work around this. You will probably have to rewrite the `RequestHandlerClass` attribute as a property, even though it isn't actually a property. If you want, it can still be a type alias, if you make the type alias generic.

Reply:
You could do it without making the `RequestHandlerClass` attribute a property:

```python
_S = TypeVar("_S", bound=BaseServer[Any])
_RequestHandlerClassCallable = Callable[[Any, Any, BaseServer], _S]

class BaseServer(Generic[_S]):
    RequestHandlerClass: _RequestHandlerClassCallable[_S]

class UDPServer(BaseServer[UDPServer]): ...
```

Reply:
Generic-only-in-typing classes tend to confuse users, so I wouldn't do this just to avoid a property.

Reply:
Yeah, fair enough. I guess you could make it a read-write property as well.

Reply:
Agree that we should change this, since this annotation would require users to supply a callback that accepts any BaseServer.